connectors/sync_job_runner.py

```diff
@@ -144,7 +144,11 @@ async def _sync_done(self, sync_status, result=None, sync_error=None):
         else:
             await self.sync_job.done(ingestion_stats=ingestion_stats)
 
-        self.sync_job = await self.sync_job.reload()
+        try:
+            await self.sync_job.reload()
+        except Exception as e:
+            logger.error(f"Failed to reload sync job {self.job_id}. Error: {e}")
+            self.sync_job = None
```

Comment:
putting `None` here seems pretty drastic, what happens next on the sync_doc call?

Reply:
~For now, this is the only place `reload` is used.~

You just reminds me I need to add `reload` in `prepare` and `validate_filtering` methods: https://github.com/elastic/connectors-python/pull/583/commits/e269d684601249972e37979c2d74cf9ce2259f66

>what happens next on the sync_doc call?

It will updates `last_sync_status` to `error`:
https://github.com/elastic/connectors-python/blob/e269d684601249972e37979c2d74cf9ce2259f66/connectors/byoc.py#L483

This is one of the edge cases we need to handle in https://github.com/elastic/connectors-python/pull/517 during job execution.

Reply:
~For now, this is the only place `reload` is used.~

You just reminds me I need to add `reload` in `prepare` and `validate_filtering` methods: https://github.com/elastic/connectors-python/pull/583/commits/e269d684601249972e37979c2d74cf9ce2259f66

>what happens next on the sync_doc call?

It will updates `last_sync_status` to `error`:
https://github.com/elastic/connectors-python/blob/e269d684601249972e37979c2d74cf9ce2259f66/connectors/byoc.py#L483

This is one of the edge cases we need to handle in https://github.com/elastic/connectors-python/pull/517 during job execution.